Make the README less misleading by putting some reasonable defaults in (instead of empty strings `''`) (#302)
julia-actions/setup-julia: README.md

    #          For example, `1.10` may be parsed as `1.1`.

>     #
>     # Default: '1'

Reviewer: ```suggestion
```

    # Note: you can use X64, X86, and ARM64 as synonyms for x64, x86, and aarch64, respectively.
    #

>     # Defaults to the architecture of the runner executing the job.

Reviewer: ```suggestion
    # Specifying default uses the architecture of the runner executing the job.
```

Author: ```suggestion
    # Specifying `'default'` uses the architecture of the runner executing the job.
```

    # never: Never print versioninfo

>     #
>     # Default: false

Reviewer: ```suggestion
```

    show-versioninfo: 'false'

    # Set the path to the project directory or file to use when resolving some versions (e.g. `min`).
    #
    # Defaults to using JULIA_PROJECT if defined, otherwise '.'

>     project: '.'

Reviewer: Should be JULIA_PROJECT?

Author: Yeah, this one is tricky, because it's hard to sum up the behavior pithily.

Using GitHub Actions syntax, I suppose it would look like this:

```yaml
project: ${{ env.JULIA_PROJECT == 'refs/heads/main' && '.' || env.JULIA_PROJECT }}
```

But I don't know if that's how we actually implement it. It seems like it'd be misleading to give an implementation in the README.

Also, the above isn't actually correct, because it doesn't distinguish between JULIA_PROJECT being unset vs being set to an empty string. So I don't think we should do that.

Reviewer: Since they are still commits, you can be semi-free form
```suggestion
    project: ${{ env.JULIA_PROJECT }} or '.' (if JULIA_PROJECT is unset)
```
